### PR TITLE
SideEffectsPlugin don't modify cache entries

### LINKED ISSUE #7400: Recompilation in development mode generates incorrect imports
# Bug report

**What is the current behavior?**

We have an import from an ES module which is compiled correctly when webpack is started, but starts failing when changes are made in other files while webpack is running in watch mode: a warning is emitted about an import that was not found and the generated code stops working. When inspecting the output file, we can see that the expected individual import has been replaced by a (non-equivalent) import of the entire module where the import is from.

The problem first appeared for us when we upgraded `date-fns` from version `2.0.0-alpha7` to `2.0.0-alpha8`, but we were able to construct a minimal reproduction project without any dependencies other than webpack itself.

**If the current behavior is a bug, please provide the steps to reproduce.**

- Unzip the attached reproduction project: [webpack-import-recompile-bug.zip](https://github.com/webpack/webpack/files/2039509/webpack-import-recompile-bug.zip)
- Run `npm i` and `npm start`
- Observe that the initial compilation does not produce any warnings
- While webpack is running, make any change to `main.js` (e.g. a comment line)
- Observe that a warning is generated on `require-me.js`:
```
WARNING in ./src/require-me.js
4:16-26 "export 'default' (imported as 'myFunction') was not found in './my-module'
 @ ./src/require-me.js
 @ ./src/main.js
```
- Now make any change to `require-me.js` (e.g. a comment line)
- Observe that compilation succeeds again

The problem only appears when running in `mode: "development"`, not `"production"`. If `sideEffects: false` is removed from `src/my-module/package.json`, or if `sideEffects: true` is removed from `webpack.config.js`, the problem also disappears.

**What is the expected behavior?**

The compilation output remains consistent while webpack is running in watch mode, without warnings about missing imports appearing and disappearing when making unrelated changes. Our only workaround is to restart webpack, which increases our wait time from ± 5 seconds to more than 20 seconds after each change.

**Other relevant information:**
webpack version: 4.8.3
Node.js version: 8.9.4
Operating System: Windows 10
Additional tools: -

### PR DESCRIPTION
**Pull request by sokra**
fix bug where modules are incorrectly including in chunks

fixes #7400

**What kind of change does this PR introduce?**
bugfix

**Did you add tests for your changes?**
yes

**Does this PR introduce a breaking change?**
yes, but its a bugfix

**What needs to be documented once your changes are merged?**
nothing

### CODE REVIEW

File: lib/FlagDependencyUsagePlugin.js [hunk 1]
```diff
@@ -86,9 +86,9 @@ class FlagDependencyUsagePlugin {
 					}
 
 					const queue = [];
-					for (const chunk of compilation.chunks) {
-						if (chunk.entryModule) {
-							processModule(chunk.entryModule, true);
+					for (const preparedEntrypoint of compilation._preparedEntrypoints) {
```
Comment (ooflorent) [inline lib/FlagDependencyUsagePlugin.js]: It is possible to use `compilation.entrypoints` instead?

Comment (sokra) [reply lib/FlagDependencyUsagePlugin.js]: Sadly not, as this plugin runs before these property is set. But we should make `_preparedEntrypoints` public in the next major version

Comment (ooflorent) [reply lib/FlagDependencyUsagePlugin.js]: Thanks for clarification. Since `_preparedEntrypoints` is a private API, isn't it safe to change it now?

Comment (sokra) [reply lib/FlagDependencyUsagePlugin.js]: Technically yes, but I better don't risk it...

**Pull request merged**